### PR TITLE
ALBS-919 Decrease Pulp storage size for backup

### PR DESCRIPTION
**Pull request by maccelf**

### CODE REVIEW

File: scripts/remove_unnecessary_versions_of_repositories.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    pulp_repos = await get_all_pulp_repositories(pulp_client, logger)
+    logger.info("Get all production repos")
+    prod_repos = await get_production_repositories(pulp_client, logger)
+    prod_repos_href = []
+    for prod_repo in prod_repos:
+        prod_repos_href.append(prod_repo.pulp_href)
+    logger.info("Removing unnecessery versions of repositories...")
+    for pulp_repo in pulp_repos:
+        if pulp_repo["pulp_href"] in prod_repos_href:
+            retain_repo_versions = 3
```
Comment (Korulag) [inline scripts/remove_unnecessary_versions_of_repositories.py:167]: I suggest to leave 3 versions for all RPM repos. That should leave enough data to be able to track all needed info for SBOM

File: scripts/remove_unnecessary_versions_of_repositories.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        else:
+            retain_repo_versions = 2
+        await remove_unnecessery_versions(
+            pulp_client=pulp_client,
+            min_retain_repo_versions=retain_repo_versions,
+            pulp_repo=pulp_repo,
+            logger=logger,
+        )
+    logger.info("All unnecessery version of repositories is deleted")
+    # await remove_orphans(pulp_client, logger)
```
Comment (Korulag) [inline scripts/remove_unnecessary_versions_of_repositories.py:177]: Remove the call (and maybe a function) if it's not used

**Review by Korulag — CHANGES_REQUESTED** [review]

**Review by Korulag — APPROVED** [review]

File: scripts/remove_unnecessary_versions_of_repositories.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    pulp_repos = await get_all_pulp_repositories(pulp_client, logger)
+    logger.info("Removing unnecessery versions of repositories...")
+    for pulp_repo in pulp_repos:
+        await remove_unnecessery_versions(
+            pulp_client=pulp_client,
+            min_retain_repo_versions=3,
+            pulp_repo=pulp_repo,
+            logger=logger,
+        )
+    logger.info("All unnecessery version of repositories is deleted")
```
Comment (javihernandez) [inline scripts/remove_unnecessary_versions_of_repositories.py:131]: "All unnecessary versions of repositories have been deleted"
Also, replace "unnecessery" with "unnecessary", found some of them

**Review by javihernandez — APPROVED** [review]
Apart from the minor comment left, LGTM

**Review by soksanichenko — APPROVED** [review]

**Pull request merged**